src/java/org/apache/cassandra/journal/Journal.java

```diff
@@ -659,6 +657,47 @@ void selectSegmentToFlush(Collection<ActiveSegment<K, V>> into)
         segments().selectActive(currentSegment.descriptor.timestamp, into);
     }
 
+    ActiveSegment<K, V> oldestActiveSegment()
+    {
+        ActiveSegment<K, V> current = currentSegment;
+        if (current == null)
```

Comment:
I don't think this can be `null`, at least as currently invoked by `Flusher`, started after we've advanced for the first time already. I don't mind the check, but if we have it here then we should probably put one in `getActiveSegment()` below as well for consistency at least?

Reply:
My feeling was that `getActiveSegment` expresses the belief that segment exists, so there must be an active segment. Whereas `oldestActiveSegment` doesn't necessarily express a belief that such a segment exists (without a `get` or `try` I suppose it is a bit ambiguous, but in particular without providing a key it is less implied).

That said, I don't have a strong feeling at all and happy to go with your preference here.